File: api-clients/javascript-closure-angular/API/Client/Strain.js
goog.provide('API.Client.Strain');

/**
 * @record
 */
API.Client.Strain = function() {}

/**
 * Name of the strain.
 * @type {!string}
 * @export
 */
API.Client.Strain.prototype.name;

/**
 * Open Cannabis Product Code for the strain.
 * @type {!string}
 * @export
 */
API.Client.Strain.prototype.ocpc;

/**
 * Name and Open Cannabis Product Code of the seed company that created the strain.
 * @type {!API.Client.Object}
 * @export
 */
API.Client.Strain.prototype.seedCompany;

/**
 * URL for QR that leads to page on Cannabis Reports.
 * @type {!string}
 * @export
 */
API.Client.Strain.prototype.qr;

/**
 * URL for strain on Cannabis Reports.
 * @type {!string}
 * @export
 */
API.Client.Strain.prototype.url;

/**
 * URL for the main photo of the strain.
 * @type {!string}
 * @export
 */
API.Client.Strain.prototype.image;

/**
 * Object of countries that this strain has lineages from. {'Country Name' => 'ISO 3166-1 Two Letter Country Code'}
 * @type {!API.Client.Object}
 * @export
 */
API.Client.Strain.prototype.lineage;

/**
 * Names of the parents of the strains and their associated Open Cannabis Product Codes.
 * @type {!API.Client.Object}
 * @export
 */
API.Client.Strain.prototype.genetics;

/**
 * Open Cannabis Product Codes of the children of this strain.
 * @type {!Array<!string>}
 * @export
 */
API.Client.Strain.prototype.children;

/**
 * Date and time record was created, UTC.
 * @type {!Date}
 * @export
 */
API.Client.Strain.prototype.createdAt;

/**
 * Date and time record was updated, UTC.
 * @type {!Date}
 * @export
 */
API.Client.Strain.prototype.updatedAt;

File: api-clients/async-scala/src/main/scala/io/swagger/client/model/Strain.scala
package io.swagger.client.model

import org.joda.time.DateTime
import java.util.UUID


case class Strain (
  name: String,  // Name of the strain.
ocpc: Option[String],  // Open Cannabis Product Code for the strain.
seedCompany: Option[Any],  // Name and Open Cannabis Product Code of the seed company that created the strain.
qr: Option[String],  // URL for QR that leads to page on Cannabis Reports.
url: Option[String],  // URL for strain on Cannabis Reports.
image: Option[String],  // URL for the main photo of the strain.
lineage: Option[Any],  // Object of countries that this strain has lineages from. {&#39;Country Name&#39; &#x3D;&gt; &#39;ISO 3166-1 Two Letter Country Code&#39;}
genetics: Option[Any],  // Names of the parents of the strains and their associated Open Cannabis Product Codes.
children: Option[List[String]],  // Open Cannabis Product Codes of the children of this strain.
createdAt: Option[DateTime],  // Date and time record was created, UTC.
updatedAt: Option[DateTime]  // Date and time record was updated, UTC.
)
